### DISCUSSION TITLE
Argument of type '{}' is not assignable to parameter of type 'SetStateAction<bigint>'

### DISCUSSION THREAD

**Discussion by a-znk** [wevm/wagmi#2890] [Q&A]
I recently upgraded to wagmi 1.3.9, viem 1.5.4 and rainbowkit 1.0.8. Now i am receiving an error i wasn't getting before:

`Argument of type '{}' is not assignable to parameter of type 'SetStateAction<bigint>'.`

Here is my code:

`const [totalSupply, setTotalSupply] = React.useState(0n);`

`const { data: totalSupplyData } = useContractRead({
    ...contractConfig,
    functionName: "totalSupply",
    watch: true,
  });`

`React.useEffect(() => {
    if (totalSupplyData) {
      setTotalSupply(totalSupplyData);
    }
  }, [totalSupplyData]);`

I'm having trouble locating the issue. I found it strange though, when i hover over `totalSupplyData` i see `const totalSupplyData: unknown` which seems wrong to me. I'm thinking i should be seeing `const totalSupplyData: bigint | undefineconst totalSupplyData: bigint | undefine`.

Somebody please help! It feels like the answer should be so obvious for me... but something isn't clicking. My code "appears" to be identical to all examples i can find.

**Comment by tmm** [comment]
What does `contractConfig` look like? Attached a complete minimal reproduction will allow folks to help you.

**Reply by a-znk** [reply]
Pardon the slow reply. I was away for a few weeks. This is what `contractConfig` looks like.

`const contractConfig = {
  addressOrName: "contract address which I'd like to keep private for now",
  contractInterface: abi,
};`

I will work on creating a minimal reproduction as well. Thank you!

**Reply by tmm** [reply]
Sounds good. Definitely need more info, like what does `abi` look like.
